### PR TITLE
Media cards

### PR DESCRIPTION
**Pull request by ruicosta042**
## What does this change?
image

It adapts the CardLink component for displaying media cards. These are defined by designType=Media. Media cards can be of three types: Audio, Video or Photo, defined on the new `mediaType`prop. It can also display a duration for Audio and Video cards. Here's the new props for `CardType`. 

```typescript
type MediaType = 'Video' | 'Audio' | 'Photo';

interface CardType {
    linkTo: string;
    pillar: Pillar;
    headline: CardHeadlineType;
    webPublicationDate?: string;
    trailImage?: CardImageType;
    standfirst?: string;
    avatar?: AvatarType;
    showClock?: boolean;
    designType?: DesignType;
    mediaType?: MediaType; // new
    mediaDuration?: number; // new
}

```

## Why?
Media cards are visually distinct:
- dark background
- icon on the right of the footer
- possible duration next to the icon
- bold font used on the footer

As defined on the visual spec for implementation on DCR: https://docs.google.com/document/d/1rWstjRUUh6-YH5XcPvOtLwsCc2nY7oMoL_JwHe-83z4/edit?usp=sharing

## Link to supporting Trello card
https://trello.com/c/nT3RRsJ4/1002-media-cards

### CODE REVIEW

File: src/web/components/Card/components/MediaMeta.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+`;
+
+export function secondsToDuration(secs?: number): string {
+    if (typeof secs === `undefined` || secs === 0) {
+        return ``;
+    }
+    const dtime = new Date(secs * 1000).toISOString();
+    const duration = secs > 3600 ? dtime.substr(11, 8) : dtime.substr(14, 5);
+    return /^0/.test(duration) ? duration.substr(1) : duration;
+}
```
Comment (oliverlloyd) [inline src/web/components/Card/components/MediaMeta.tsx]: This is longer but removes the need for string manipulation:
```typescript
export function secondsToDuration(secs?: number): string {
    if (typeof secs === `undefined` || secs === 0) {
        return ``;
    }

    const seconds = Number(secs);
    var h = Math.floor(seconds / 3600);
    var m = Math.floor((seconds % 3600) / 60);
    var s = Math.floor((seconds % 3600) % 60);

    let duration = [];
    if (h > 0) {
        duration.push(h);
    }
    if (m > 0 || h === 0) {
        // supports 0:59
        duration.push(m);
    }
    if (s > 0) {
        duration.push(s);
    }
    return duration.join(':');
}
```
What do you think?

File: src/web/components/Card/components/MediaMeta.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        transform: ${mediaType === 'Video' ? `translateY(0.0625rem)` : ``};
+    }
+`;
+
+const durationStyles = ({ pillar }: Props) => css`
+    color: ${palette[pillar].main};
+    font-size: 0.75rem;
+    line-height: 1rem;
+    ${textSans.xsmall()}
+    font-weight: 600;
```
[the remark this answers is not in the document]
Comment (ruicosta042) [reply src/web/components/Card/components/MediaMeta.tsx]: hey, `textSans` supports `fontWeight`already

File: src/web/components/Card/components/CardFooter.tsx [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
     display: flex;
+    justify-content: space-between;
+    align-items: center;
 `;
 
-export const CardFooter = ({ children }: Props) => (
-    <footer className={footerStyles}>{children}</footer>
+const mediaFooter = css`
+    flex-direction: row-reverse;
+    padding: 0 0.3125rem 0.3125rem 0.3125rem;
```
[the remark this answers is not in the document]
Comment (ruicosta042) [reply src/web/components/Card/components/CardFooter.tsx]: hey, converted all `rems` introduced on this PR to `px`

File: src/web/components/Card/components/CardLink.tsx [hunk 4]
```diff
@@ -39,6 +39,18 @@ const linkStyles = (designType: DesignType | undefined) => {
...
             `;
         case 'Media':
+            return css`
+                ${baseLinkStyles}
+                background-color: ${palette.neutral[20]};
+               :hover {
+                     /* TODO: This colour is hard coded here because it does not yet
+                           exist in src-foundation. Once it's been added, please
+                           remove this. @siadcock is aware. */
+                    /* stylelint-disable-next-line color-no-hex */
```
[the remark this answers is not in the document]
Comment (ruicosta042) [reply src/web/components/Card/components/CardLink.tsx]: removed

File: src/web/components/Card/components/MediaMeta.tsx (hunk 1 above)
Comment (ruicosta042) [reply src/web/components/Card/components/MediaMeta.tsx]: I've a sweet spot for one-liners and concise functions, but your version is much more idiomatic and semantically sound. Changed.

**Comment by paperboyo** [comment]
image
😍 💎

File: src/web/components/Card/components/MediaMeta.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+`;
+
+export function secondsToDuration(secs?: number): string {
+    if (typeof secs === `undefined` || secs === 0) {
+        return ``;
+    }
+    const seconds = Number(secs);
+    const h = Math.floor(seconds / 3600);
+    const m = Math.floor((seconds % 3600) / 60);
+    const s = Math.floor((seconds % 3600) % 60);
```
Comment (mxdvl) [inline src/web/components/Card/components/MediaMeta.tsx:52]: This is equivalent to `const s = Math.floor(seconds % 60);` as 3600 is divisible by 60!

**Pull request merged**